### PR TITLE
[USM] Fix get all blocked process debug endpoint

### PR DESCRIPTION
**Pull request by amitslavin**
### What does this PR do?

Fixes a bug in the blocked process debug endpoint that affects the agent status as well.

### Motivation

Fix a bug to include the correct blocked process for the debug endpoint. This PR affects the endpoint in two ways:

* When there are no blocked PIDs, the program name will not be shown at all in the `blocked_processes` section.

### Additional Notes

**Example of the endpoint result before the change:**
```Blocked Processes: [
      {
        "PathIdentifiers": [],
        "ProgramType": "shared_libraries"
      },
      {
        "PathIdentifiers": [
          {
            "Dev": 64768,
            "Inode": 398470,
            "SamplePath": "/proc/1/root/usr/lib/systemd/systemd"
          },
          {
            "Dev": 64768,
            "Inode": 398484,
            "SamplePath": "/proc/443/root/usr/lib/systemd/systemd-journald"
          },
          .
          .
          .
]
```
**After the change:**
```
Blocked Processes: [
  {
    "ProgramType": "shared_libraries",
    "PathIdentifiers": [
      {
        "Dev": 64768,
        "Inode": 413168,
        "SamplePath": "/proc/36403/root/lib/aarch64-linux-gnu/libssl.so.3"
      }
    ]
  },
  {
    "ProgramType": "go-tls",
    "PathIdentifiers": [
      {
        "Dev": 64768,
        "Inode": 398484,
        "SamplePath": "/proc/442/root/usr/lib/systemd/systemd-journald"
      },
      {
        "Dev": 64768,
        "Inode": 407458,
        "SamplePath": "/proc/481/root/usr/sbin/multipathd"
      },
      .
      .
      .
]      
```  

### Describe how to test/QA your changes

To do:

**Benchmark results:** 
 my branch: BenchmarkGetUSMStats-4   	 5507845	       213.4 ns/op
 main:           BenchmarkGetUSMStats-4   	 6427843	       176.3 ns/op
 
 As well deployed on staging cluster and did not observe any negative impact. 

QA steps:
Enable system-probe with TLS monitoring:
```yaml
service_monitoring_config:
  enabled: true
  tls:
    go:
      enabled: true
    native:
      enabled: true
  enable_http_monitoring: true
```

Step 1: Run the agent

Step 2: Run agent status command

Expect to see

```bash
============
System Probe
============

  Status: Running
  Uptime: 209ns
  Last Updated: 2024-07-11 06:05:07 PDT / 2024-07-11 13:05:07 UTC (1720703107000)

  USM
  ===
    Status: running
    Last Check: 2024-07-11 06:05:04 PDT / 2024-07-11 13:05:04 UTC (1720703104000)
    Traced Programs: [
      {
        "FilePath": "/usr/lib/aarch64-linux-gnu/libcrypto.so.3",
        "PIDs": [
          44672,
          442,
          801,
          38280,
          40305,
          40458,
          41612
        ],
        "ProgramType": "shared_libraries"
      },
      {
        "FilePath": "/usr/lib/aarch64-linux-gnu/libgnutls.so.30.31.0",
        "PIDs": [
          22165,
          723
        ],
        "ProgramType": "shared_libraries"
      },
      {
        "FilePath": "/usr/bin/dockerd",
        "PIDs": [
          954
        ],
        "ProgramType": "go-tls"
      }
    ]
    Blocked Processes: [
      {
        "PathIdentifiers": [
          {
            "Dev": 64768,
            "Inode": 398484,
            "SamplePath": "/proc/442/root/usr/lib/systemd/systemd-journald"
          },
          {
            "Dev": 64768,
            "Inode": 407458,
            "SamplePath": "/proc/481/root/usr/sbin/multipathd"
          },
          {
            "Dev": 64768,
            "Inode": 405991,
            "SamplePath": "/proc/486/root/usr/bin/udevadm"
          },
          {
            "Dev": 64768,
            "Inode": 397805,
            "SamplePath": "/proc/683/root/usr/lib/systemd/systemd-timesyncd"
          },
          {
            "Dev": 64768,
            "Inode": 398490,
            "SamplePath": "/proc/721/root/usr/lib/systemd/systemd-networkd"
          },
          {
            "Dev": 64768,
            "Inode": 398497,
            "SamplePath": "/proc/723/root/usr/lib/systemd/systemd-resolved"
          },
          {
            "Dev": 64768,
            "Inode": 407355,
            "SamplePath": "/proc/740/root/usr/sbin/cron"
          },
          {
            "Dev": 64768,
            "Inode": 393913,
            "SamplePath": "/proc/741/root/usr/bin/dbus-daemon"
          },
          {
            "Dev": 64768,
            "Inode": 407417,
            "SamplePath": "/proc/745/root/usr/sbin/irqbalance"
          },
          {
            "Dev": 64768,
            "Inode": 407001,
            "SamplePath": "/proc/748/root/usr/libexec/polkitd"
          },
          {
            "Dev": 64768,
            "Inode": 407488,
            "SamplePath": "/proc/750/root/usr/sbin/rsyslogd"
          },
          {
            "Dev": 1797,
            "Inode": 568,
            "SamplePath": "/proc/756/root/snap/snapd/21761/usr/lib/snapd/snapd"
          },
          {
            "Dev": 64768,
            "Inode": 398486,
            "SamplePath": "/proc/759/root/usr/lib/systemd/systemd-logind"
          },
          {
            "Dev": 64768,
            "Inode": 410335,
            "SamplePath": "/proc/783/root/usr/sbin/ModemManager"
          },
        ],
        "ProgramType": "go-tls"
      }
    ]

  NPM
  ===
    Status: Running
    Last Check: 2024-07-11 06:05:02 PDT / 2024-07-11 13:05:02 UTC (1720703102000)

  Event Monitor
  ================
    Status: Running

  Process
  =======
    Status: Running


========================
Transport Proxy Warnings
========================

  No Transport Proxy Warnings
```
Step 3: Validate that all of the programs types inside the `Blocked Processes` part, are not empty.

### CODE REVIEW

File: pkg/network/usm/utils/debugger.go [hunk 1]
```diff
@@ -170,9 +173,6 @@ func (d *tlsDebugger) GetBlockedPathIDsWithSamplePath(programType string) []Path
 			continue
 		}
 
-		registry.m.Lock()
-		defer registry.m.Unlock()
-
```
Comment (guyarb) [inline pkg/network/usm/utils/debugger.go:175]: why did you remove the lock?

Comment (amitslavin) [reply pkg/network/usm/utils/debugger.go]: returned the lock

**Review by guyarb — APPROVED** [review]
